backend/Application/Api/GraphQL/Transactions/TransactionResultEvent.cs

```diff
@@ -181,6 +188,63 @@ internal static IEnumerable<TransactionResultEvent> ToIter(ContractUpdateIssued
         }
     }
 
+    /// <summary>
+    /// Try parse hexadecimal events from module schema.
+    /// </summary>
+    protected string[]? GetParsedEvents(
+        ModuleReferenceEvent moduleReferenceEvent,
+        string contractName,
+        string[] eventsAsHex,
+        ILogger logger
+        )
+    {
+        if (moduleReferenceEvent.Schema == null)
+        {
+            return null;
+        }
+        var events = new string[eventsAsHex.Length];
+        for (var i = 0; i < eventsAsHex.Length; i++)
+        {
+            try
+            {
+                var eventAsHex = eventsAsHex[i];
+                var eventContract = InteropBinding.GetEventContract(moduleReferenceEvent.Schema, contractName, eventAsHex, moduleReferenceEvent.SchemaVersion);
+                if (eventContract == null)
+                {
+                    logger.Warning("{ContractName} on {Module} got null returned when parsing hexadecimal event {EventAsHex}", contractName, moduleReferenceEvent.ModuleReference, eventAsHex);
+                    return null;
+                }
+                events[i] = eventContract;
+            }
+            catch (InteropBindingException e)
+            {
+                switch (e.Error)
```

Comment:
I feel like I saw the exact same switch elsewhere. Perhaps there could be some code reuse for the handling of the InteropBindingException?

Reply:
There is one in `ReceiveName` which also handles interop errors, but the logic is different.

Ex. `NoReceiveInContrac`are not expected when deserializing events, and this should trigger a `logger.Error` in this case.